apps/s_cb.c

```diff
@@ -1373,27 +1428,27 @@ static int security_callback_debug(const SSL *s, const SSL_CTX *ctx,
         {
             const unsigned char *salg = other;
             const char *sname = NULL;
-            switch (salg[1]) {
-            case TLSEXT_signature_anonymous:
-                sname = "anonymous";
-                break;
-            case TLSEXT_signature_rsa:
-                sname = "RSA";
-                break;
-            case TLSEXT_signature_dsa:
-                sname = "DSA";
-                break;
-            case TLSEXT_signature_ecdsa:
-                sname = "ECDSA";
-                break;
-            }
+            int raw_sig_code= (salg[0]<<8) + salg[1]; /* always big endian (msb, lsb) */
+                /* raw_sig_code: signature_scheme from tls1.3, or signature_and_hash from tls1.2 */
 
-            BIO_puts(sdb->out, OBJ_nid2sn(nid));
-            if (sname)
-                BIO_printf(sdb->out, ", algorithm=%s", sname);
-            else
-                BIO_printf(sdb->out, ", algid=%d", salg[1]);
-            break;
+            if (nm != NULL) BIO_printf(sdb->out, "%s", nm);
+            else            BIO_printf(sdb->out, "s_cb.c:security_callback_debug op=0x%x", op);
```

Comment:
Style nit: Should be like this

````
if (nm != NULL)
    xxx;
else
    yyy;
````
i.e. the the if and else branches need to be on separate lines

Reply:
Done